src/main/java/com/cburch/logisim/std/io/DotMatrixHDLGeneratorFactory.java

```diff
@@ -57,32 +57,60 @@
 
     contents.add("  ");
     if (colBased) {
-      for (var thisRow = 0; thisRow < rows; thisRow++)
-        for (var thisCol = 0; thisCol < cols; thisCol++) {
-          final var wire = (rows == 1) ? GetNetName(componentInfo, thisCol, true, netlist) 
-              : GetBusEntryName(componentInfo, thisCol, true, thisRow, netlist);
-          final var idx = thisRow * cols + thisCol + componentInfo.GetLocalBubbleOutputStartId();
+      /* The simulator uses here following addressing scheme (2x2):
+       *  r1,c0 r1,c1
+       *  r0,c0 r0,c1
+       *  
+       *  hence the rows are inverted to the definition of the LED-Matrix that uses:
+       *  r0,c0 r0,c1
+       *  r1,c0 r1,c1
+       */
+      for (var dotMatrixRow = 0; dotMatrixRow < rows; dotMatrixRow++) {
+        final var ledMatrixRow = rows - dotMatrixRow - 1;
+        for (var ledMatrixCol = 0; ledMatrixCol < cols; ledMatrixCol++) {
+          final var wire = (rows == 1) ? GetNetName(componentInfo, ledMatrixCol, true, netlist) 
```

Comment:
These looped ternariy assignments look pretty similar here in 3 loop blocks. Any chance to extract this code to separate method and reduce overall LoC?

Reply:
As the code is already the minimal, I could think of making an other loop and do the switch inside the loop, but I am not sure this would improve readability. For the moment I think the code is as condense at it can be, adding another file for this will not resolve the nr. of code lines. So I prefer that it stays this way.